src/web/routes/application/steps/address/select-address/select-address.js

```diff
@@ -34,9 +37,11 @@ const resetAddressState = (req) => {
 
 const behaviourForGet = (config, journey) => (req, res, next) => {
   resetAddressState(req)
+  const addressId = req.session.claim.addressId
   res.locals.postcodeLookupError = req.session.postcodeLookupError
   if (!res.locals.postcodeLookupError) {
-    res.locals.addresses = req.session.postcodeLookupResults.map(buildAddressOption)
+    res.locals.addresses = req.session.postcodeLookupResults.map(buildAddressOption(addressId))
+    res.locals.addressSelected = res.locals.addresses.filter(addr => addr.selected).length > 0
```

Comment:
Might be better to use `Array.some()` here:
```
const isSelected = addr => addr.selected
res.locals.addresses.some(isSelected)
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
done